hwW2_d2/src/main/kotlin/SportCar.kt

```diff
@@ -0,0 +1,19 @@
+class SportCar(override var type: String, startEngin: Boolean, speed: Int, gear: Int):Vehicle(type,startEngin,speed,gear) {
+
+
+    override fun changeGear(num: Int) {
+if (startEngin==true){
+    gear=(1..5).random()
```

Comment:
You didn’t pass the num parameter to the function, instead, you set it randomly.